modules/caas/api/src/main/java/io/cattle/platform/api/service/ServiceCreateValidationFilter.java

```diff
@@ -367,14 +368,30 @@ protected void validateIpsHostName(ApiRequest request) {
         }
     }
 
-    protected void validateLaunchConfigs(Service service, ApiRequest request) {
+    @SuppressWarnings("unchecked")
+    protected ApiRequest validateLaunchConfigs(Service service, ApiRequest request) {
         Map<String, Object> data = CollectionUtils.toMap(request.getRequestObject());
+
+        List<Object> modifiedSlcs = new ArrayList<>();
+        if (data.get(ServiceConstants.FIELD_SECONDARY_LAUNCH_CONFIGS) != null) {
+            List<Object> slcs = (List<Object>) data.get(ServiceConstants.FIELD_SECONDARY_LAUNCH_CONFIGS);
+            for (Object slcObj : slcs) {
+                Map<String, Object> slc = (Map<String, Object>) slcObj;
+                slc.put(InstanceConstants.FIELD_NETWORK_MODE, NetworkConstants.NETWORK_MODE_CONTAINER);
```

Comment:
The network mode needs to be set to host or none if the primary is set to host or none.  Also, the primary can not be "--net=container".

Additionally you need to validate standalone container.  That can be in a different PR.